compiler/luci/pass/src/FoldCastPass.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldCastPass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+template <typename FromT, typename ToT> ToT safe_down_cast(FromT from)
```

Comment:
I think this should not be applied.
When I executed following cmds, `safe_down_cast` is not we expected.
```python
import tensorflow as tf

a = tf.constant(21000000000, dtype=tf.int64)
b = tf.cast(a, dtype=tf.int32)
print(b) # tf.Tensor(-453397504, shape=(), dtype=int32)
```

Reply:
Do you prefer not to check overflow just like TF?

Reply:
Um... I think it would be better to get some advice from App part guys...

Reply:
Of course, if overflow is occurred in some model, most of it might be unintended.

However, even if overflow is unintended result, I think we should not change original graph.
The most important role of compiler frontend is **generating (optimized)circle model which is exactly same with original model.**
i.e., Introducing `safe_down_cast` means that `out1` and `out2` is same but it is not.

```python
import tensorflow as tf

inp = tf.constant(21000000000, dtype=tf.int64)
out1 = tf.cast(a, dtype=tf.int32)

min = tf.minimum(INT32_MAX, inp)
max = tf.maximum(INT32_MIN, min)
out2 = tf.cast(max, dtype=tf.int32)
```

Therefore, for me, `safe_down_cast` seems not proper :)

Reply:
> I think it would be better to get some advice from App part guys...

This is about how to handle a corner case, which does not happen in our target models, so App part may not be related to this issue. The casted value in our target model is dimension, which hardly exceeds int32 boundary.

Reply:
@llFreetimell Thanks for the comment :) It seems that I was a bit late to remove the comment.